src/eth/types.ts

```diff
@@ -11,3 +13,19 @@ export const EthUint256Struct = definePattern(
   'EthUint256',
   /^0x([1-9a-f][0-9a-f]*|0)$/iu,
 );
+
+export const BundlerUrlStruct = refine(
+  string(),
+  'BundlerUrl',
+  (value: string) => {
+    let url;
+
+    try {
+      url = new URL(value);
+    } catch (_) {
+      return false;
+    }
+
+    return url.protocol === 'http:' || url.protocol === 'https:';
+  },
```

Comment:
Can you add an error message? The default message is a bit confusing:

```
Expected a value of type `string`, but received: `"random string"`
```

Something like can be a bit more clear:

```
Expected a valid URL, but received: ${value}
```